### ISSUE TITLE
Follow system theme by default

### ISSUE THREAD

**Issue by aha999** [fr3ts0n/AndrOBD#128]
I think AndrOBD should follow system theme by default (If the system theme is dark, AndrOBD is automatically dark at startup, if it's light, AndrOBD is light). Of course there would be a manual setting, but by default is should be on 'follow system theme' in my opinion.

- Light
- Dark
- System default (the recommended default option)

More in this link:
https://developer.android.com/guide/topics/ui/look-and-feel/darktheme

**Comment by fr3ts0n** [comment]
This would break backwards compatibility!
- System themes were introduced with Android 10
- AndrOBD supports Dark/Light mode (manually selected) down to Android 4.2

**Comment by aha999** [comment]
Well, then if it supports it, it should follow the system theme, but if not, it would be auto-set to light mode.
Can it detect android version and hide the follow system theme if android version is older than android 10 ?
I mean, there are apps that can follow a system theme and also support older versions of android... So i don't know how that feature is implemented, nor am i a dev.

**Comment by willhubs** [comment]
> Well, then if it supports it, it should follow the system theme, but if not, it would be auto-set to light mode and have a manual switch toggle in the settings.

I agree with this. this would be the ideal situation in my opinion.

**Comment by RokeJulianLockhart** [comment]
As aforestated at https://github.com/fr3ts0n/AndrOBD/issues/104#issuecomment-2501200243:

> Is there a consensus that any new GUI design would be Material 3 + You? I would consider anything that doesn't adhere to it to be automatically redundant, considering how much it integrates with AOSP ≥ 12. It would remediate https://github.com/fr3ts0n/AndrOBD/issues/128#issue-697059386.

**Comment by chrisdebian** [comment]
Revisiting this now that Phase 1 of the Material migration (#104) has landed. Worth a fresh look since the original objection was about backward compatibility, and that's changed:

Since #335, minSdk is now 23 (Android 6.0), not 4.2 — Android's system-level dark theme needs API 29 (Android 10), so pre-29 devices would have nothing to "follow" either way, but that's a much smaller gap than the original 4.2 concern.

One thing worth flagging before this becomes a "just add it" ticket: the app's day/night switching isn't built on `AppCompatDelegate.setDefaultNightMode()` — it's a fully custom mechanism (`MainActivity.nightMode`, a boolean preference, manual `setTheme()` calls). Adding a genuine "follow system" option means picking one of two different shapes:

1. **Migrate to `AppCompatDelegate`'s built-in day/night modes** (`MODE_NIGHT_FOLLOW_SYSTEM`/`MODE_NIGHT_YES`/`MODE_NIGHT_NO`) — the "correct" long-term approach, but touches every activity's theme-setting code, a real refactor not a quick toggle.
2. **Add "System default" as a third option to the existing boolean preference**, checking `Configuration.uiMode` at startup only — much smaller change, but won't live-update if the user switches system theme while the app is open (would need `onConfigurationChanged` for that, which the current custom mechanism doesn't have hooked up anywhere).

Happy to scope either properly once there's a steer on which shape you'd prefer — or if this should just wait and ride along whenever the day/night system gets touched for other reasons.

**Comment by fr3ts0n** [comment]
Sounds to me like the 1st option is the proper way to do it. So let's do it the proper way.
Thx

**Comment by beedell-roke** [comment]
@chrisdebian, although accurate, try to disclaim when you entirely author a message via an LLM.

**Comment by chrisdebian** [comment]
Fair point, thanks for flagging. That comment was drafted with Claude based on a review of the code, then edited and posted by me. Happy to flag this on substantive analysis comments going forward too, not just PR descriptions where CONTRIBUTING.md currently asks for it.

**Comment by chrisdebian** [comment]
Opened #354.
